input/input.go

```diff
@@ -69,6 +71,10 @@ type TxInfo struct {
 	Weight int64
 }
 
+func (t *TxInfo) String() string {
```

Comment:
fixed the docs!

> Probably just worth doing if there actually are situations where TxInfo could be nil.

Good question. I think the string method is only meant to be used in logging with `%v`, so if it's nil then it will be printed as `nil`.  Plus the name `TxInfo` is a bit misleading, what it really means is, the parent tx info for a given input, which has limited use cases. So I think it's fine to leave it as is.